### ISSUE TITLE
hook is not of type Query

### ISSUE THREAD

**Issue by frederik-b** [Automattic/mongoose#7790]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
this context in hooks are not of type Query anymore

**If the current behavior is a bug, please provide the steps to reproduce.**

query hook:
```
mySchema.pre('findOneAndUpdate', async function setDefaultOptions(this: any) {
  console.log(this instanceof Query) // false, used to be true up to mongoose 5.4.22
})
```
call findOneAndUpdate
```
await model(message.messageType).findOneAndUpdate({_id: {$exists: false}}, newMessage, {upsert: true})
````
**What is the expected behavior?**

behaviour should not change between minor versions of mongoose
query hooks should be of type Query

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
mongoose 5.5.6 is broken (and earlier)
mongoose 5.4.22 is the last tested working version

```
# mongod --version                       
db version v4.0.9
```

**Comment by vkarpov15** [comment]
The below script runs as expected for me, `this instanceof mongoose.Query === true` :

```javascript
const assert = require('assert');
const mongoose = require('mongoose');
mongoose.set('debug', true);
mongoose.set('useFindAndModify', false);

const GITHUB_ISSUE = `gh7790`;
const connectionString = `mongodb://localhost:27017/${ GITHUB_ISSUE }`;
const { Schema } = mongoose;

run().then(() => console.log('done')).catch(error => console.error(error.stack));

async function run() {
  await mongoose.connect(connectionString);
  await mongoose.connection.dropDatabase();

  const mySchema = new Schema({ name: String });

  mySchema.pre('findOneAndUpdate', async function setDefaultOptions() {
    console.log(this instanceof mongoose.Query); // true
  });

  const Model = mongoose.model('Test', mySchema);

  await Model.findOneAndUpdate({_id: {$exists: false}}, { name: 'test' }, {upsert: true});
}
```

Can you modify the above script to demonstrate your issue? Also, why does your function accept `this` as a parameter? That might contribute to the issue, although I'm not familiar with TypeScript.

**Comment by frederik-b** [comment]
I managed to create a (rather large) test script by reducing everything to the absolute minimum untill I found the problematic code section.
It is definitely https://www.npmjs.com/package/mongoose-delete, which breaks it.

I created a repo with my "reduced" test setup here: https://github.com/frederik-b/mongoosetest .

So it's not a bug with mongoose itself, it's rather that mongoose-delete breaks something which it didn't break in the previous versions.

**Comment by vkarpov15** [comment]
This issue comes down to how mongoose-delete is implemented: they overwrite built-in model functions using custom statics, and we [added support for hooks for custom statics in Mongoose 5.5](http://thecodebarbarian.com/mongoose-5-5-static-hooks-and-populate-match-functions.html). So this means you'd get hooks for 'findOneAndUpdate()' with `this` set to the model.

**Comment by vkarpov15** [comment]
Fix will be in 5.5.12
